scripts/orb-registration-script/orb-registration.py

```diff
@@ -0,0 +1,754 @@
+        shutil.move(str(self.build_dir / "uid.pub"), str(jet_artifacts_dir / "uid.pub"))
+
+        # Save orb name and token
+        with open(jet_artifacts_dir / "orb-name", "w") as f:
+            f.write(orb_name)
+        with open(jet_artifacts_dir / "token", "w") as f:
+            f.write(token)
+
+        # Copy persistent images
+        self.logger.info(
+            f"Copying base persistent images into artifacts directory for {orb_id}"
+        )
+        shutil.copy(
+            str(self.build_dir / "persistent.img"),
+            str(jet_artifacts_dir / "persistent.img"),
+        )
+        shutil.copy(
+            str(self.build_dir / "persistent-journaled.img"),
+            str(jet_artifacts_dir / "persistent-journaled.img"),
+        )
+
+        # Mount and install orb-name and token in both images
+        for img_name in ["persistent.img", "persistent-journaled.img"]:
+            self.logger.info(f"Mounting {img_name} for Orb ID: {orb_id}")
+            subprocess.run(
+                ["mount", str(jet_artifacts_dir / img_name), str(mount_point)],
+                check=True,
+            )
+
+            try:
+                subprocess.run(
+                    [
+                        "install",
+                        "-o",
+                        "0",
+                        "-g",
+                        "0",
+                        "-m",
+                        "644",
+                        str(jet_artifacts_dir / "orb-name"),
+                        str(mount_point / "orb-name"),
+                    ],
+                    check=True,
+                )
+                subprocess.run(
+                    [
+                        "install",
+                        "-o",
+                        "0",
+                        "-g",
+                        "0",
+                        "-m",
+                        "644",
+                        str(jet_artifacts_dir / "token"),
+                        str(mount_point / "token"),
+                    ],
+                    check=True,
+                )
+                subprocess.run(["sync"], check=True)
+            finally:
+                subprocess.run(["umount", str(mount_point)], check=True)
+
+    def process_pearl_orb(self, cf_token: str, mount_point: Path) -> str:
+        """Process a single Pearl orb (generate ID, register, create artifacts)."""
+        orb_id = self.generate_orb_id()
+        platform = self.detect_platform(self.args.hardware_version)
+
+        orb_name = self.register_orb_mongo(orb_id, cf_token, platform)
+        self.set_orb_channel(orb_id, cf_token)
+        token = self.get_orb_token(orb_id, cf_token)
+
+        self.save_orb_artifacts(orb_id, orb_name, token, mount_point)
+        self.register_orb_core_app(orb_id, orb_name)
+
+        return orb_id
+
+    def process_diamond_orb_ids(self, orb_ids: List[str], cf_token: str):
```

Comment:
this is not a good design - you duplicate code both for platforms and also both for slightly different input args formats(e.g pairs vs ids), probably something very simple like base class would reduce some code duplication. but i comment here only just fyi and learning, i don't think we should spend a lot of time on this util script